Source/WebCore/page/DOMWindow.cpp

```diff
@@ -139,4 +139,42 @@ WebCoreOpaqueRoot root(DOMWindow* window)
     return WebCoreOpaqueRoot { window };
 }
 
+WindowProxy* DOMWindow::opener() const
+{
+    RefPtr frame = this->frame();
+    if (!frame)
+        return nullptr;
+
+    RefPtr localFrame = dynamicDowncast<LocalFrame>(frame);
+    if (localFrame && !localFrame->loader().opener())
+        return nullptr;
+
+    return &frame->windowProxy();
```

Comment:
This is not correct.  Also this function doesn't need a dynamicDowncast<LocalFrame> because Frame::opener exists.

Reply:
Whoops, will switch this to the opener window proxy.